Quaver.Shared/Screens/Gameplay/Rulesets/Keys/GameplayRulesetKeys.cs

```diff
@@ -58,10 +63,128 @@ internal static bool IsDownscroll
         /// </summary>
         /// <param name="screen"></param>
         /// <param name="map"></param>
-        public GameplayRulesetKeys(GameplayScreen screen, Qua map) : base(screen, map)
+        public GameplayRulesetKeys(GameplayScreen screen, Qua map) : base(screen, map) => InitializeTimingLines();
+
+        /// <summary>
+        ///     Initialiize Timing Line Manager(s) depending on Scroll Direction.
+        /// </summary>
+        private void InitializeTimingLines()
         {
+            // Create appropriate Timing Line Managers.
+            // - Multiple Timing Line Managers for multiple Scroll Directions
             if (ConfigManager.DisplayTimingLines.Value)
-                TimingLineManager = CreateTimingLineManager();
+            {
+                var playfield = (GameplayPlayfieldKeys)Playfield;
+                switch (MapManager.Selected.Value.Qua.Mode)
+                {
+                    case GameMode.Keys4:
+                        {
+                            switch (ConfigManager.ScrollDirection4K.Value)
+                            {
+                                case ScrollDirection.Down:
```

Comment:
This code doesn't look good visually. Just use one line or break it into two lines if it's too long.